tutorials/cloud-iot-logging/index.md

```diff
@@ -0,0 +1,238 @@
+
+
+      // write the log entryto Stackdriver Logging
+      const entry = log.entry(metadata, logData);
+      return log.write(entry);
+    }
+  });
+```
+
+To deploy the cloud function, we use the Firebase CLI tool:
+
+```sh
+cd functions
+npm install
+firebase use $GCLOUD_PROJECT
+firebase deploy --only functions
+```
+
+## Write logs from a device
+
+create a dummy sample device:
+
+```sh
+cd ../sample-device
+npm install
+
+gcloud iot devices create log-tester --region $CLOUD_REGION --registry $REGISTRY_ID --public-key path=./ec_public.pem,type=ES256
+
+node build/index.js &
+```
+
+Note: do not use this device for any real workloads, as the keypair is included in this sample and should not be considered secret.
+
+## Explore the logs that are written
+
+If you open up the <a href="https://console.cloud.google.com/logs/viewer" target="_blank">Stackdriver Logging console</a>.
+
+![console image](./images/c1.png)
+
+Because we send the device id as part of the log record, you can choose to pull that up into the summary line:
+
+![console image](./images/c2.png)
+
+Which then looks like this:
+
+![console image](./images/c3.png)
+
+### Combine system and Applications logs
+
+Now we will exercise a part of our sample device code that responds to config changes. Use the following gcloud command to update the devices config telling it to "bounce to a level of 2":
+
+```sh
+gcloud iot devices configs update --device log-tester --registry $REGISTRY_ID --region $CLOUD_REGION --config-data '{"bounce": 2}'
+```
+
+Now in just a few moments, you will see two new entries in the logging console. One is from IoT Core system noting that a devices config was updated (the ModifyCloudToDeviceConfig call).
+
+This is then followed by a device application log reporting the imaginary "spring back" value. This shows how we can view both system logs from IoT Core and device application logs in one place.
+
+You can use the refresh button in the Cloud Console, or use the play button to stream logs.
+
+![console image](./images/c4.png)
+
+### Use severity to convey log level
+
+Lets send a couple more config updates (you should send one, wait a second or two, then send the next):
+
+```sh
+gcloud iot devices configs update --device log-tester --registry $REGISTRY_ID --region $CLOUD_REGION --config-data '{"bounce": 7}'
+
+# pause
+
+gcloud iot devices configs update --device log-tester --registry $REGISTRY_ID --region $CLOUD_REGION --config-data '{"bounce": 17}'
+
+```
+
+As these are received and responded to by the sample device, you can see the use of a severity level to indicate the importance of the log.
+
+![console image](./images/c5.png)
+
+### Filter logs to a specific device
+
+We currently have been looking at a log that contains entries for all devices.  But given these are structured log entries, we can use the numeric id of the resource to limit our view to only one device in the more realistic scenario when multiple devices are writing log events.
```

Comment:
Use the second person as much as possible. This is particularly important when giving direct instructions or suggesting an action the reader would take. I would rewrite this paragraph something like this to make this clearer:

So far in this tutorial, you have worked with logs containing entries for all devices. You can limit your view to a single device by using its numeric device ID.

Open the log entry and find the resource. Go to the labels and click on `device_num_id`. Choose `Show matching entries`.